guides/specification.md

```diff
@@ -0,0 +1,555 @@
+# Trento Checks DSL Specification
+
+A language allowing to declare best practices to be adhered on target SAP Infrastructures.
+
+# Table of contents
+
+- [Introduction](#introduction)
+- [Anatomy of a Check](#anatomy-of-a-check)
+  - [Filename Convention](#filename-convention)
+  - [Structure](#structure)
+    - [id](#id)
+    - [name](#name)
+    - [group](#group)
+    - [description](#description)
+    - [remediation](#remediation)
+    - [severity](#severity)
+    - [Facts, Values, Expectations](#facts-values-expectations)
+- [Facts](#facts)
+- [Values](#values)
+- [Expectations](#expectations)
+- [Expression Language](#expression-language)
+    - [Evaluation Scope](#evaluation-scope)
+
+## Introduction
+
+The need this DSL aims to fulfill is to provide users a simple way to declare what we (the Trento Core Team) often refer to as "Checks". 
+
+Checks are, in Trento's domain, the crystallization of SUSE's best practices when it comes to SAP workloads in a form that both a man and a machine can read.
+
+We get several benefits from this approach:
+
+- Users can formalize best practices with no space for ambiguity;
+- Machines can assert systems' state, automatically, with no space for ambiguity;
+- The development of new best practices gets streamlined through a common definition that allows to firestart shared efforts.
+
+## Anatomy of a Check
+
+A Check declaration comes in the form of a `yaml` file and all the Checks together build up the **Checks Catalog**
+
+Here's an example:
+```yaml
+id: 156F64
+name: Corosync `token` timeout
+group: Corosync
+description: |
+  Corosync `token` timeout is set to `{{ values.expected_token_timeout }}`
```

Comment:
I would remove this renderization in the description. 
We don't have it (and I honestly thing we shouldn't have it).
Let's play safe